### ISSUE TITLE
Tracking Issue: Incremental PR for Primitive Types in Proptests

### ISSUE THREAD

**Issue by YoshikiTakashima** [model-checking/kani#1473]
We will merge into a feature branch for code review. 

- [ ] #1516: CI and infra
  - Blocked on license issue.
- [ ] TBD: Strategy, runner and macro
- [ ] TBD: Arbitrary
- [ ] TBD: booleans and tuples
- [ ] TBD: `proptest::prelude`
- [ ] TBD: numbers
- [ ] TBD:

**Comment by YoshikiTakashima** [comment]
Replaced by #1608
